lib/Dialect/TritonGPU/Transforms/Pipeliner/MatmulLoopPipeline.cpp

```diff
@@ -400,13 +400,19 @@ static bool loadIsMMAv3(Operation *loadOp) {
 static llvm::MapVector<Operation *, LoadInfo>
 assignMemoryLayouts(llvm::SmallVector<std::tuple<Operation *, int, Operation *>>
                         &loadOpToIndLevelAndUse,
-                    tt::ModuleAxisInfoAnalysis &axisInfoAnalysis) {
+                    tt::ModuleAxisInfoAnalysis &axisInfoAnalysis,
+                    int numStages) {
   llvm::MapVector<Operation *, LoadInfo> loadToInfo;
 
   for (auto &[op, dist, use] : loadOpToIndLevelAndUse) {
     if (loadToInfo.count(op))
       // TODO pawel: err, we'd need to verify that the distance is the same
       continue;
+    if (numStages == 2 && dist > 0)
```

Comment:
1) Perhaps it would be good to not do it in `assignMemoryLayouts` to not introduce another piece unrelated to lowering? We can do this analysis before this call.
2) Can we make it more generic, like `if (dist > numStages-1) continue;`?